app/controllers/staff/speakers_controller.rb

```diff
@@ -1,36 +1,26 @@
 class Staff::SpeakersController < Staff::ApplicationController
   decorates_assigned :speaker
-  before_filter :set_proposal, only: [:new, :create, :destroy]
+  before_filter :set_program_session, only: [:new, :create, :destroy]
 
   def index
-    render locals: {
-             proposals: @event.proposals.includes(speakers: :user).decorate
-           }
+    @program_session_speakers = Speaker.with_program_session.for_event(current_event)
```

Comment:
What about an approach directly off of the current event such as  current_event.speakers?